### DISCUSSION TITLE
Responsibility for session lifespan management

### DISCUSSION THREAD

**Discussion by davidoram** [ory/kratos#1207] [Q&A] [answered]
Hi, I'm working on a [blog post](https://github.com/davidoram/kratos-selfservice-ui-go/blob/main/blog-post/Blog-post.md) that describes integrating a Go application with Krytos.

One thing isn't clear to me is the recommended best practice for protecting routes that require that the user is logged in. I've started with [middleware](https://github.com/davidoram/kratos-selfservice-ui-go/blob/main/middleware/kratos_auth.go) that calls the `WhoAmI` API on **every** request.   My thinking is that then  Krytos can expire the session as per its config (`session.lifespan`). 

Is that the recommended approach?

**Comment by aeneasr** [comment, marked as answer]
Yes, that is recommended! We’re also working on a proxy which has some caching built in to make this stuff less costly. However it is the only way to ensure that when your users sign out they are signed out!

**Reply by davidoram** [reply]
Many thanks @aeneasr 👍
